src/classes/frDonation.cls

```diff
@@ -133,18 +136,29 @@ public class frDonation extends frModel {
                                             frUtil.Entity.SUBSCRIPTION, subscriptionId);
             }
         }
+        Opportunity opp = getOpportunity();
+        
         
+        Boolean isPledge = Boolean.valueOf(request.get('pledge'));
+        if(!isPledge) {
+            Pledge__c pledge = frPledge.findActive(opp.fr_Donor__c);
```

Comment:
I think this will need to change.  If an update is sent from Funraise for an old donation that was linked to a pledge, this line would find that there isn't an active pledge anymore and unlink it.  So I think we'll only want to try this frPledge.findActive call if and only if there isn't already a pledge linked to the opportunity, which means we'll probably need to query for it

Reply:
Ok now the boolean alreadyMappedToPledge handles this